src/tweets.ts

```diff
@@ -154,25 +156,55 @@ export function getTweetsByUserId(
   });
 }
 
+export async function getTweetWhere(
+  options: TweetMatchOptions,
+  tweets: AsyncGenerator<Tweet> | Tweet[],
+): Promise<Tweet | null> {
+  for await (const tweet of tweets) {
+    const matches = Object.keys(options).every((k) => {
+      const key = k as keyof Tweet;
+      return tweet[key] == options[key];
+    });
+
+    if (matches) {
+      return tweet;
+    }
+  }
+
+  return null;
+}
+
+export async function getTweetsWhere(
+  options: TweetMatchOptions,
+  tweets: AsyncGenerator<Tweet> | Tweet[],
```

Comment:
I think there's something like `Iterable<T>` for this, I don't remember which TS targets have that, though.

Reply:
Should be [`AsyncIterator<T>`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncIterator), which I think supports sync iteration as well?